src/Servers/Kestrel/Core/src/KestrelBadHttpRequestException.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core
+{
+    internal class KestrelBadHttpRequestException : IOException
```

Comment:
It might be easier to have this derive from Kestrel's BadHttpRequestException.

Reply:
I'm not sure why this copy of BadHttpRequestException even exists?

Reply:
That's true, if we're going to be throwing the old type anyway, we can just keep the internal stuff there for now.